File: SDKs/googleapis.com/securitycenter/v1/python/src/sdk/models/shared/cve.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import cvssv3 as shared_cvssv3
from ..shared import reference as shared_reference


@dataclass_json
@dataclasses.dataclass
class Cve:
    r"""Cve
    CVE stands for Common Vulnerabilities and Exposures. More information: https://cve.mitre.org
    """
    
    cvssv3: Optional[shared_cvssv3.Cvssv3] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cvssv3') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    references: Optional[list[shared_reference.Reference]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('references') }})
    upstream_fix_available: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('upstreamFixAvailable') }})
    

File: SDKs/googleapis.com/securitycenter/v1/java/lib/src/main/java/openapisdk/models/shared/Cve.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Cve
 * CVE stands for Common Vulnerabilities and Exposures. More information: https://cve.mitre.org
**/
public class Cve {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cvssv3")
    public Cvssv3 cvssv3;
    public Cve withCvssv3(Cvssv3 cvssv3) {
        this.cvssv3 = cvssv3;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public Cve withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("references")
    public Reference[] references;
    public Cve withReferences(Reference[] references) {
        this.references = references;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("upstreamFixAvailable")
    public Boolean upstreamFixAvailable;
    public Cve withUpstreamFixAvailable(Boolean upstreamFixAvailable) {
        this.upstreamFixAvailable = upstreamFixAvailable;
        return this;
    }
}